# Resumo e Decisões

## Parte 1:
Esses são os pontos que vou retirar, setembro de 2021, maio de 2024 e dezembro de 2025. Os dois primeiros baseados na média geral e o último baseado na média do respectivo mês nos últimos 4 anos



### Parte 1

Importar as bibliotecas e carregar os dados de geração

In [ ]:
import pandas as pd

# 1. Carregar os dados
tabela_usina = pd.read_excel('power_plant_data_teste.xlsx')

# verificar se tem linhas vazias
display(tabela_usina.info())

# Mostrar a tabela
display(tabela_usina)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 61 entries, 0 to 60
Data columns (total 3 columns):
 #   Column                                     Non-Null Count  Dtype         
---  ------                                     --------------  -----         
 0   Geração Mensal Referência Month            60 non-null     datetime64[ns]
 1   Unidade Consumidora (UC) Usina (Nickname)  61 non-null     object        
 2   Geração Mensal SUM Energia Gerada (kWh)    61 non-null     int64         
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 1.6+ KB


None

,Geração Mensal Referência Month,Unidade Consumidora (UC) Usina (Nickname),Geração Mensal SUM Energia Gerada (kWh)
0,NaT,Usina Teste,0
1,2021-07-01,Usina Teste,374203
2,2021-08-01,Usina Teste,404653
3,2021-09-01,Usina Teste,426355
4,2021-10-01,Usina Teste,467200
...,...,...,...
56,2026-02-01,Usina Teste,438651
57,2026-03-01,Usina Teste,363221
58,2026-04-01,Usina Teste,364546
59,2026-05-01,Usina Teste,306272


In [ ]:
#Temos uma linha vazia na tabela! Limpando essa linha
tabela_usina = tabela_usina.dropna()
display(tabela_usina.info())

display(tabela_usina.describe().round(1))

<class 'pandas.core.frame.DataFrame'>
Index: 60 entries, 1 to 60
Data columns (total 3 columns):
 #   Column                                     Non-Null Count  Dtype         
---  ------                                     --------------  -----         
 0   Geração Mensal Referência Month            60 non-null     datetime64[ns]
 1   Unidade Consumidora (UC) Usina (Nickname)  60 non-null     object        
 2   Geração Mensal SUM Energia Gerada (kWh)    60 non-null     int64         
dtypes: datetime64[ns](1), int64(1), object(1)
memory usage: 1.9+ KB


None

,Geração Mensal Referência Month,Geração Mensal SUM Energia Gerada (kWh)
count,60,60.0
mean,2023-12-16 11:12:00,417105.5
min,2021-07-01 00:00:00,110463.0
25%,2022-09-23 12:00:00,352583.5
50%,2023-12-16 12:00:00,422588.0
75%,2025-03-08 18:00:00,494157.2
max,2026-06-01 00:00:00,730252.0
std,NaN,101307.7


In [12]:
import plotly.express as px

# Gráfico de barras simples: Mês no eixo X e Geração no eixo Y
fig = px.bar(
    tabela_usina, 
    x='Geração Mensal Referência Month', 
    y='Geração Mensal SUM Energia Gerada (kWh)',
    title='Geração Mensal da Usina (kWh)',
    labels={'data': 'Mês/Ano', 'geracao': 'Geração (kWh)'}
)

fig.show()


### Importante:
    Temos uma média de geração de 417105 kWh no período avaliado
    Setembro de 2021 tem a maior geração com 730252 kWh, aproximadamente 75% mais produção que a media
    Maio de 2024 tem a menor geração com 110463 kWh, aproximadamente 26% da média
    Dezembro de 2025 está com geração de 266 kWh, sendo aproximadamente 50% da média dos últimos 4 anos (21 a 24 => média aproximada 519 kwh)

Esses são os pontos que vou retirar, setembro de 2021, maio de 2024 e dezembro de 2025. Os dois primeiros baseados na média geral e o último baseado na média do respectivo mês nos últimos 4 anos


### Parte 2